File: app/src/main/java/statusbar/finder/sql/ActiveManager.java
package statusbar.finder.sql;

import statusbar.finder.ActiveQueries;
import statusbar.finder.DatabaseHelper;
import statusbar.finder.OriginQueries;

/**
 * LyricGetterExt - statusbar.finder.sql
 *
 * @author VictorModi
 * @description TODO: coming soon.
 * @email victormodi@outlook.com
 * @date 2025/1/20 下午5:16
 */
public class ActiveManager {
    private static ActiveQueries queries = null;

    public static Long insertActiveLog(Long originId, Long resultId) {
        if (queries == null) {queries = DatabaseHelper.getDatabase().getActiveQueries();}
        queries.insertActive(originId, resultId);
        return queries.getLastInsertId().executeAsOne();
    }

    public static void updateActiveLog(Long originId, Long resultId) {
        if (queries == null) {queries = DatabaseHelper.getDatabase().getActiveQueries();}
        queries.updateActive(originId, resultId);
    }

    public static Long getResultIdByOriginId(Long originId) {
        if (queries == null) {queries = DatabaseHelper.getDatabase().getActiveQueries();}
        return queries.getResultId(originId).executeAsOneOrNull();
    }
}

File: app/src/main/java/statusbar/finder/sql/ActiveManager.kt
package statusbar.finder.sql

import statusbar.finder.ActiveQueries
import statusbar.finder.DatabaseHelper

/**
 * LyricGetterExt - statusbar.finder.sql
 * @description TODO: coming soon.
 * @author VictorModi
 * @email victormodi@outlook.com
 * @date 2025/2/9 13:53
 */
object ActiveManager {
    private val queries: ActiveQueries by lazy {
        DatabaseHelper.database?.activeQueries
            ?: throw IllegalStateException("Database not initialized")
    }

    fun insertActiveLog(originId: Long, resultId: Long) {
        queries.insertActive(originId, resultId)
    }

    fun updateActiveLog(originId: Long, resultId: Long) {
        queries.updateActive(originId, resultId)
    }

    fun getResultIdByOriginId(originId: Long): Long? {
        return queries.getResultId(originId).executeAsOneOrNull()
    }
}

